ui/js/component/overlayRouter/view.jsx

```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import ShowPage from "page/show";
+
+class OverlayRouter extends React.PureComponent {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const { currentPage, params, playingUri, keepPlaying } = this.props;
+
+    const uri = params.uri || playingUri;
+    const isPage = currentPage === "show";
+
+    const page = uri && (keepPlaying || isPage)
+      ? <ShowPage expanded={isPage} uri={uri} />
```

Comment:
The `ShowPage` should not be doing double duty of being both a full page and the overlay component. Allowing them to each be their own function will be a much cleaner design.